internal/backend/retry/backend_retry.go

```diff
@@ -126,10 +126,16 @@ func (be *Backend) Save(ctx context.Context, h restic.Handle, rd restic.RewindRe
 // is returned. rd must be closed after use. If an error is returned, the
 // ReadCloser must be nil.
 func (be *Backend) Load(ctx context.Context, h restic.Handle, length int, offset int64, consumer func(rd io.Reader) error) (err error) {
-	return be.retry(ctx, fmt.Sprintf("Load(%v, %v, %v)", h, length, offset),
+	err = be.retry(ctx, fmt.Sprintf("Load(%v, %v, %v)", h, length, offset),
 		func() error {
-			return be.Backend.Load(ctx, h, length, offset, consumer)
+			var innerError error
+			innerError = be.Backend.Load(ctx, h, length, offset, consumer)
```

Comment:
`err := ...` would also work here as it declares a new variable that shadows the outer `err`.